src/newmatrix.cpp

```diff
@@ -1039,8 +1037,29 @@ void matrix_fight(struct matrix_icon *icon, struct matrix_icon *targ)
     }
   } // else { implementations of slow, etc, would go here }
 
-  // Results of damaging attacks
-  success -= success_test(bod, power);
+  if (targ->type == ICON_LIVING_PERSONA && ICON_IS_IC(icon) && icon->ic.type >= IC_LETHAL_BLACK) {
+    // For black IC we'll be be doing our ICCM checks up front
+    // since we bypass doing a (separate) damage pass from the condition monitor of the persona
+    // and instead directly attack the otaku
+    int resist = 0;
+    bool lethal = icon->ic.type == IC_LETHAL_BLACK ? TRUE : FALSE;
+    if (!targ->decker->asist[0] && lethal)
+      lethal = FALSE;
```

Comment:
This can probably be skipped since the living persona is always treated as having a hot ASIST.